### DISCUSSION TITLE
Nodemcu and GC9A01

### DISCUSSION THREAD

**Discussion by kiruhcs** [Bodmer/TFT_eSPI#1490] [Q&A - General]
Hello,

First Bodmer THANKS for your work!
I have some issues with my display and nodemcu.
This is my connection layout 
![image](https://user-images.githubusercontent.com/78826209/146084650-2dd7112b-033b-4e64-8a49-8a9d4b26061a.png)
I edited the User_Setup.h -> uncommented this:
#define GC9A01_DRIVER and 
#define TFT_HEIGHT 240 // GC9A01 240 x 240
So everything should be fine? 
But if I load an example like (drawXBitmap or something else) to the Nodemcu nothing happen.

I hope someone can help me :)
Best regards and many thanks
Alex

**Comment by Bodmer** [comment]
Run the "Read_User_Setup" sketch and check the output on the Serial monitor window to verify the setup picked up be the compiler are correct.

**Reply by kiruhcs** [reply]
This is the output from the "Read_User_Setup". 

`[code]
TFT_eSPI ver = 2.3.70
Processor    = ESP8266
Frequency    = 80MHz
Voltage      = 3.31V
Transactions = No
Interface    = SPI
SPI overlap  = No

Display driver = 9A01
Display width  = 240
Display height = 240

TFT_CS   = PIN_D8
TFT_DC   = PIN_D3

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 27.00
[/code]
`

**Reply by Bodmer** [reply]
Looks like you have not defined the TFT_RST pin as D4.

**Reply by strebor76** [reply]
> This is the output from the "Read_User_Setup".
> 
> `[code] TFT_eSPI ver = 2.3.70 Processor = ESP8266 Frequency = 80MHz Voltage = 3.31V Transactions = No Interface = SPI SPI overlap = No
> 
> Display driver = 9A01 Display width = 240 Display height = 240
> 
> TFT_CS = PIN_D8 TFT_DC = PIN_D3
> 
> Font GLCD loaded Font 2 loaded Font 4 loaded Font 6 loaded Font 7 loaded Font 8 loaded Smooth font enabled
> 
> Display SPI frequency = 27.00 [/code] `

Hi Kiruchs.

How did you go with this? I am having the same issue, but don't seem to be able to get the screen to work. I've ran the "read use setup" sketch, but nothing comes out on the serial monitor. Sketch does say it has successfully upload though.

Thanks.

Jonno.

**Reply by Bodmer** [reply]
Make sure the Serial Monitor window baud rate is set to 115200.

**Reply by strebor76** [reply]
Cool. Got that to work. Below is my output:


[code]
TFT_eSPI ver = 2.4.2
Processor    = ESP8266
Frequency    = 80MHz
Voltage      = 3.29V
Transactions = No
Interface    = SPI
SPI overlap  = No

Display driver = 9A01
Display width  = 240
Display height = 240

TFT_CS   = PIN_D8
TFT_DC   = PIN_D3
TFT_RST  = PIN_D4

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 27.00
[/code]

**Reply by kiruhcs** [reply]
Hi strebor76,

I have still issues with the ESP8266. I tried now the NodeMCU ESP32 and the display works fine.
Can't figure out the problem with the ESP8266.

**Comment by Kotik82** [comment]
I know i am too late, but maybe it helps someone else in the future, the issue with esp8266 and this specific display is that the CS pin is by default pulled HIGH from the display's level converter and as a result if u connect the display to D8  (GPIO15) then nothing works since that PIN needs to be LOW during ESP8266 booting. Just pick a different PIN for CS and make sure the one u pick can be pulled HIGH during boot.

**Reply by Bodmer** [reply]
Thanks for reporting this, I expect this is why the OP had problems.

**Reply by arunneoito** [reply]
Thanks ! i got my display working with this information

**Reply by slihp67** [reply]
"Just pick a different PIN for CS and make sure the one u pick can be pulled HIGH during boot." how do you do this?
